### ISSUE TITLE
using as a gateway

### ISSUE THREAD

**Issue by salihy** [davehorton/drachtio-b2b-media-proxy#2]
Is it possible to use this as a gateway? What I need is when there is 2 seperate networks and my server has 2 interface for each network, one interface will communicate to trunk for both signaling(drachtio) and media(rtpengine) and other interface will communicate with sip clients for both signaling and media and it will route traffic between those legs? I tried to do it, I added 2 interfaces, I also configured routes but rtpengine tries to send rtp from the original interface (since there is no routing, it cannot send it).

You can find pcap file attached:  [log19.zip](https://github.com/davehorton/drachtio-b2b-media-proxy/files/2230944/log19.zip)

![image](https://user-images.githubusercontent.com/4982519/43248675-484d77e8-90c1-11e8-948a-450b1d5ec21d.png)

**Comment by davehorton** [comment]
yes it should be possible.  You would need to add two 'contact' elements in your drachtio.conf.xml, one for each local address to bind to.  Have you tried that?  If so can you show me that part of your config?

**Comment by salihy** [comment]
I think there is nothing wrong with drachtio, drachtio communicates correctly, but rtpengine does not. You can see the log file attached.
```

        <contacts>
            <contact>sip:*;transport=udp,tcp</contact>
            <!-- local sip address to bind to.  Default: 'sip:*', which means listens on port 5060 on all interfaces and transports -->
            <!--
                    Other examples:
                        sip:192.168.1.100        listen on a specific address
                        sip:*;transport=tcp      tcp connections only
                        sip:*:5061               listen on port 5061
            -->


            <!-- here's how to listen for every protocol on their default ports -->
<!--
            <contact>sip:*:80;transport=ws</contact>
            <contact>sips:*;transport=tls</contact>
            <contact>sips:*:443;transport=wss</contact>
-->

        </contacts>
```

**Comment by salihy** [comment]
I think I will need to use direction propert of rtpengine offer function somehow, do you have any idea how to do that?

**Comment by davehorton** [comment]
I think you would configure multiple interfaces in rtpengine, as described [here](https://github.com/sipwise/rtpengine#interfaces-configuration).

Have you tried that?  How exactly is it failing?  I also find it useful to turn log levels to debug for rtpengine and review its log output in /var/log/messages

**Comment by salihy** [comment]
I literally tried everything, I configured multiple interfaces, rtpengine configuration, routes. I also tried to send direction parameter to rtpengine.offer function, you can find my code in the link, line 176. there is something wrong, I send sdp message correctly but rtp traffic doesnt start,

Here is my pcap log, note that 0.87 is my internal ip, 227.170 is external ip, 127.20 telco sip server, 127.44 is teco rtp server, 0.144 is local asterisk which I send incoming call to: 
[log20.zip](https://github.com/davehorton/drachtio-b2b-media-proxy/files/2241172/log20.zip)


https://github.com/salihy/Drachtio.B2BSipServer/blob/master/app.js
